src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java

```diff
@@ -263,73 +372,142 @@ public String getRemote() {
+                int number = ghPullRequest.getNumber();
+                listener.getLogger().format("%n    Checking pull request %s%n", HyperlinkNote.encodeTo(ghPullRequest.getHtmlUrl().toString(), "#" + number));
+                boolean fork = !repo.getOwner().equals(ghPullRequest.getHead().getUser());
+                if (fork && !buildForkPRMerge && !buildForkPRHead) {
+                    listener.getLogger().format("    Submitted from fork, skipping%n%n");
                     continue;
                 }
-            }
-            SCMHead head = new SCMHead(branchName);
-            SCMRevision hash = new SCMRevisionImpl(head, entry.getValue().getSHA1());
-            observer.observe(head, hash);
-            if (!observer.isObserving()) {
-                return;
-            }
-            branches++;
-        }
-        listener.getLogger().format("%n  %d branches were processed%n", branches);
-
-        listener.getLogger().format("%n  Getting remote pull requests...%n");
-        int pullrequests = 0;
-        for (GHPullRequest ghPullRequest : repo.getPullRequests(GHIssueState.OPEN)) {
-            PullRequestSCMHead head = new PullRequestSCMHead(ghPullRequest);
-            final String branchName = head.getName();
-            listener.getLogger().format("%n    Checking pull request %s%n", HyperlinkNote.encodeTo(ghPullRequest.getHtmlUrl().toString(), "#" + branchName));
-            if (repo.getOwner().equals(ghPullRequest.getHead().getUser())) {
-                listener.getLogger().format("    Submitted from origin repository, skipping%n%n");
-                continue;
-            }
-            if (criteria != null) {
-                SCMSourceCriteria.Probe probe = getProbe(branchName, "pull request", "refs/pull/" + head.getNumber() + "/head", repo, listener);
-                if (criteria.isHead(probe, listener)) {
-                    // FYI https://developer.github.com/v3/pulls/#response-1
-                    Boolean mergeable = ghPullRequest.getMergeable();
-                    if (Boolean.FALSE.equals(mergeable)) {
-                        listener.getLogger().format("      Not mergeable, but it will be included%n");
-                    }
-                    listener.getLogger().format("    Met criteria%n");
-                } else {
-                    listener.getLogger().format("    Does not meet criteria%n");
+                if (!fork && !buildOriginPRMerge && !buildOriginPRHead) {
+                    listener.getLogger().format("    Submitted from origin repository, skipping%n%n");
                     continue;
                 }
+                if (!fork) {
+                    originBranchesWithPR.add(ghPullRequest.getHead().getRef());
+                }
+                boolean trusted = isTrusted(repo, ghPullRequest);
+                if (!trusted) {
+                    listener.getLogger().format("    (not from a trusted source)%n");
+                }
```

Comment:
And it's ok to continue when not trusted i.e. don't want to break out from here?